cmd/goplugin-heroku/resource/app.go

```diff
@@ -0,0 +1,96 @@
+package resource
+
+import (
+	"context"
+
+	"github.com/hashicorp/go-hclog"
+	heroku "github.com/heroku/heroku-go/v5"
+	"github.com/lyraproj/servicesdk/serviceapi"
+)
+
+type App struct {
+	ID           string
+	Locked       *bool
+	Maintenance  bool
+	Name         *string
+	Organization *string
+	Personal     bool
+	Region       *string
+	Space        string
+	Stack        *string
+}
+
+type AppHandler struct{}
+
+func (*AppHandler) Create(desiredState *App, service *heroku.Service) (*App, string, error) {
```

Comment:
@thallgren The core question I was trying to figure out yesterday was how to pass the second parameter into these CRUD functions. 

I have the code that will create the `heroku.Service` struct here: https://github.com/lyraproj/lyra/pull/313/files#diff-474632c35402f0d2e903f116ed76012bR31 but not sure how to implement that code correctly with the resource itself. It'll be something I'd like to understand for any API-based resource/provider that someone would want to created.

Reply:
The signature is fixed. You can't pass a second parameter here. The actual service here is perhaps something that you would want the `AppHandler` struct to contain?

All data that is passed to the CRUD functions must be expressed as attributes in your `App` resource or be part of the external ID string (this string may well be a URI with parameters baked into it, such as bearer token etc.).

Reply:
If I add that to the `AppHandler` struct, would it look something like this in `start.go`:

```go
sb.RegisterHandler("Heroku::AppHandler", &resource.AppHandler{ Client: herokuService() }, evs[0])
```

Reply:
Yes.